build/flaws.js

```diff
@@ -497,10 +499,25 @@ async function fixFixableFlaws(doc, options, document) {
           timeout: 10000,
           retry: 3,
         });
+        const fileType = await FileType.fromBuffer(imageBuffer);
+        if (!fileType) {
+          throw new Error(
+            `No file type could be extracted from ${flaw.src} at all. Probably not going to be a valid image file.`
+          );
+        }
+        if (!VALID_MIME_TYPES.has(fileType.mime)) {
+          throw new Error(
+            `${flaw.src} has an unrecognized mime type: ${fileType.mime}`
+          );
+        }
+        const imageBasename = `${path.basename(
+          decodeURI(url.pathname),
+          path.extname(decodeURI(url.pathname))
+        )}.${fileType.ext}`;
         const destination = path.join(
           Document.getFolderPath(document.metadata),
           path
-            .basename(decodeURI(url.pathname))
+            .basename(imageBasename)
             .replace(/\s+/g, "_")
```

Comment:
Side note related to https://github.com/mdn/yari/issues/2643, but I think all non-letter and digit characters should be escaped to `_`, and trim any multiple `_` to a single

Reply:
Maybe
```suggestion
            .replace(/[^\d\w\.-]/g, "_")
```

Reply:
Lemme try that!

Reply:
Cool! I was able to make this: https://github.com/mdn/content/pull/1639
which was somewhat unrelated but I'm glad I stumbled in. Thanks for pointing that out!

Reply:
Definitely a better approach than regexing it. Might make sense to pass the `options.replacement` to it to use the `_` there too

Reply:
Oh yikes! I didn't realize the `sanitizeFilename` doesn't treat whitespace. I'll fix that.

Reply:
The way that `options.replacement` works seems to be that you can pick what the replacement character can be. It doesn't mean you can say that `\s+` is considered bad. Having a filename with spaces in the name is *sane* so that's why they (i.e. `sanitizeFilename`) doesn't worry about it. Replacing whitespace for `_` is *our* worry.

Reply:
Yup, you fixed it already, my comment was more so some of the strings didn't "fold down" as it replaced the non-space characters with an empty string